api/src/main/java/org/apache/iceberg/types/EdgeInterpolationAlgorithm.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.types;
+
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/** The algorithm for interpolating edges. */
+public enum EdgeInterpolationAlgorithm {
```

Comment:
This is a long name to use as a prefix. What about `EdgeAlgorithm` instead?

Also, I'm not sure that "interpolation" is the right word for this because the algorithm calculates points on the edge but does not define the edge. The CRS defines the interpolation for the edge, right?

Reply:
I've renamed it to `EdgeAlgorithm` in the last commit, as the section name for the definitions of these algorithms is Edge-Interpolation Algorithm.